### ISSUE TITLE
Once rendered and unmounted, new renders gives blank space

### ISSUE THREAD

**Issue by gchehab** [terrestris/react-geo#1216]
I am starting an application with react-geo, and have been observing an odd behavior. 

I followed the HOC sample from the [workshop](https://terrestris.github.io/react-geo-ws/higher-order-components/mappify.html) to create an mapiffied component and rendered a instance of it right on the home/start page.  

Whenever the user navigates to other views that unmount the map provider, when they return to the home/start page, the space where the map provider was, is being rendered blank. 

Inspecting the code, the \<div class=ol-viewport\> that was supposed to be under the inner \<div class=map\> is missing on the second map rendering.

Did I miss something? Tried to force update with no avail...  Using react-geo 11.4.3, OL 5.3.3 and react 16.8.3

**Comment by marcjansen** [comment]
Thanks for the report. Do you have a minimal example that shows this behaviour?

**Comment by gchehab** [comment]
Thanks for the prompt response, I suppose that it may be something related to the react-routers I am using. The application opens a map on the Home route, when you go to About and Home routes again, the ol-viewport is not included again on under the map \<div\>

I have created a simple app at codesamplebox.io that replicates the problem:

[https://codesandbox.io/s/react-geo-sample-with-simple-router-twj80](https://codesandbox.io/s/react-geo-sample-with-simple-router-twj80?fontsize=14)

I am also copying here the sample code I have used:

example.js:
```
import React, { Component } from "react";
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import OlMap from "ol/Map";
import OlView from "ol/View";
import OlLayerTile from "ol/layer/Tile";
import OlSourceOsm from "ol/source/OSM";
import {
  MapComponent,
  MapProvider,
  mappify,
  onDropAware
} from "@terrestris/react-geo";
const Map = mappify(onDropAware(MapComponent));
const layer = new OlLayerTile({
  source: new OlSourceOsm(),
  name: "OSM"
});
const center = [0, 0];
const map = new OlMap({
  view: new OlView({
    center: center,
    zoom: 4
  }),
  layers: [layer]
});
map.on("postcompose", map.updateSize);
class LayeredMap extends Component {
  render() {
    return (
      <MapProvider map={map}>
        <Map style={{ width: 50, height: 50 }} />
      </MapProvider>
    );
  }
}
function BasicExample() {
  return (
    <Router>
      <div>
        <ul>
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/about">About</Link>
          </li>
        </ul>
        <hr />
        <Route exact path="/" component={Home} />
        <Route path="/about" component={About} />
      </div>
    </Router>
  );
}
function Home() {
  return (
    <div>
      <h2>Home</h2>
      <LayeredMap />
    </div>
  );
}
function About() {
  return (
    <div>
      <h2>About</h2>
    </div>
  );
}
export default BasicExample;
```

index.js:
```
import React from 'react';
import ReactDOM from 'react-dom';
import App from './example';
ReactDOM.render(
<App />,
document.getElementById('root')
);
```

package.json:
```
{
  "name": "React Router - Basic",
  "version": "0.0.0",
  "description": "A simple example deployed using react-codesandboxer",
  "main": "index.js",
  "dependencies": {
    "@terrestris/react-geo": "^11.4.3",
    "antd": "3.14.1",
    "babel-plugin-import": "1.12.0",
    "ol": "~5.0",
    "react": "latest",
    "react-dom": "latest",
    "react-router-dom": "latest",
    "react-scripts": "2.0.0"
  }
}
```

index.html:
`<div id="root"></div>`

**Comment by dnlkoch** [comment]
Thanks @gchehab for your valuable report.

I could trace back the error and fixed in #1218. As soon as the fix has been merged and released, your application should work as expected again.

**Comment by gchehab** [comment]
Great!  Thanks @dnlkoch, I have applied the fix on my local copy and it is working great.

**Comment by marcjansen** [comment]
Thanks for the feedback.

**Comment by dnlkoch** [comment]
The fix is available by [11.4.4](https://github.com/terrestris/react-geo/releases/tag/v11.4.4) now.
